server/matchmaker/map_pool.py

```
@@ -24,39 +26,96 @@ def __init__(
+        return adjusted_weights
+
+    def choose_map(self, played_map_ids: Iterable[int] = (), initial_weights: Optional[dict[int, float]] = None) -> Map:
+        """
+            Selects a random map using veto system weights with an anti-repetition adjustment.
         """
         if not self.maps:
-            self._logger.critical(
-                "Trying to choose a map from an empty map pool: %s", self.name
-            )
+            self._logger.critical("Trying to choose a map from an empty map pool: %s", self.name)
             raise RuntimeError(f"Map pool {self.name} not set!")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)